### DISCUSSION TITLE
exporting local map to mobile version

### DISCUSSION THREAD

**Discussion by Mitsuki-y** [ppy/osu#30256] [Ideas]
Hello, I would like to know if the function of exporting beatmaps in the mobile version has already come, or if it is still too early for this function, since I have some maps created in Osu Lazer on mobile and I cannot export them, and the export function has existed for a long time on the PC and without it the maps are impossible to share.

**Comment by Mitsuki-y** [comment]
![Screenshot_20241012_211106](https://github.com/user-attachments/assets/a30b15cf-8679-4139-825e-2be1721bcca2)
As you can see I have beatmaps made but I can't export them

**Comment by Joehuu** [comment]
For now as a workaround, you would have to run the data from your mobile device on your PC, which is a tedious process:
- connect mobile device to PC
	- on iOS, trust device and use iTunes or Finder (macOS) and go to Files/File Sharing
	- on Android, change USB preferences to file transfer and use file explorer and go to `Internal shared storage\Android\data\sh.ppy.osulazer\files`
- copy `files` folder (that has the one letter/digit folders) and `client.realm` to PC
- on PC, go to your `osu` data folder, which is the location when you press `Open osu! folder` in-game
- rename that folder to `osu PC` or something
- create a folder called `osu` or the name that was previously the PC data folder and put the mobile `files` folder and `client.realm`
- run lazer and it should have the mobile data and beatmaps that you can export via the editor as `.olz` and move those exported files somewhere else
- revert changes to get PC data back
